Supply mule should now take the actual amount that could be bought by…

Galaxy-Spam opened the pull request:
… the station into account when buying from NPCs

partically fixes #46

Galaxy-Spam committed:
Supply mule should now take the actual amount that could be bought by the station into account when buying from NPCs

bhayden53 commented:
Were you able to test this in your save?

Galaxy-Spam commented:
yes, did test it before, it worked out.
However it only fixes a single case!

bhayden53 commented:
Yep, I know, I will go through and edit all trades with the AI to also check the affordable amount. It will be pushed to this PR and then merged. Thanks for all the help.

Galaxy-Spam commented:
Thank you for the mod ;)

Misunderstood-Wookiee commented:
Suppose I'd put out a new release later today then.

On Sat, 2 May 2020, 1:37 am Galaxy-Spam, <notifications@github.com> wrote:

> Thank you for the mod ;)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Misunderstood-Wookiee/Mules-and-Warehouses-Extended/pull/47#issuecomment-622437573>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFH3SZ4NFJYG7YLGHOOIXS3RPLUBZANCNFSM4MXFKA3A>
> .
>

Galaxy-Spam committed:
Using clamping trade amount to ensure the ship has enough money available to buy resources before issuing the trade order.
Otherwise, if the staion has not enough credits, this will result in an endless loop in the script.
Fixes #46

Galaxy-Spam committed:
introduced a single variable for DebugFileName

Galaxy-Spam committed:
introduced a single variable for DebugFileName

Galaxy-Spam committed:
Merge branch 'fix-supply-mule-money' of https://github.com/Galaxy-Spam/Mules-and-Warehouses-Extended into fix-supply-mule-money

Galaxy-Spam committed:
minor comments added

Galaxy-Spam committed:
applied fix to all required cases

bhayden53 commented:
I don't think there's going to be a release ready today, I have a lot of work I need to do to get a few things ready. Nothing game breaking has shown up IMO so I don't see a need to go through the annoyance of putting out a release. 

There's no specific issue that is yet fully fixed anyway. But I am working on the issues that have piled up this weekend, so we'll see how far I get.

Galaxy-Spam committed:
added some debug info, fixed some debug strings and added debugDirName variable, fixed a logbook entry, no changes to the AI script

Galaxy-Spam committed:
Added logbookEntryTitle variable and separated logbook title from debuglog name

bhayden53 commented on aiscripts/supply_mule.xml:
```diff
@@ -5113,11 +5116,11 @@
 									+' at ' +$bestNeedTrade.unitprice.formatted.{'%s %Cr'} + ' for a profit of ' +$bestProfit.formatted.{'%s %Cr'} " />
 	
-									<write_to_logbook category="upkeep" title="'SupplyMule- ' + this.ship.knownname + ' ' + this.ship.idcode" interaction="showonmap" money="$bestProfit" text="'buying ' +$bestAmount + ' ' +$bestNeedTrade.ware +' from ' +$bestSupplyTrade.owner.knownname
+									<write_to_logbook category="upkeep" title="$debugFileName" interaction="showonmap" money="$bestProfit" text="'buying ' +$bestAmount + ' ' +$bestNeedTrade.ware +' from ' +$bestSupplyTrade.owner.knownname
```
We may actually want the ship name in the logbook entries.

Galaxy-Spam replied:
Ah, mass replace... Let me check that

Galaxy-Spam replied:
should be seperated now, see commit https://github.com/Misunderstood-Wookiee/Mules-and-Warehouses-Extended/pull/47/commits/d30d289c8f7b4d402fc69a274b3163de21e18b5e

bhayden53 commented:
I found the mod that was breaking my dev environment so I've got this running in my save now. Want to keep an eye on it for a little while before merging. Just want to see it run for a couple of hours and check the logs. Thanks for showing me their string formatting and also for clamp_trade_amount. There's probably a lot of convenience functions that could replace alot of code. One thing at a time though...

Galaxy-Spam commented:
yeah, no problem. I just started modding X4 when this problem showed up for me (3 days ago), so i had to learn all that stuff xD.

bhayden53 commented:
I just started a couple of weeks ago myself, I apparently didn't take enough time to explore the functions :)

bhayden53 commented:
This looks fine to me, only issues in my log from running overnight are other things that I know about.
